cardinal/ecs/world_logger_test.go

```diff
@@ -0,0 +1,96 @@
+package ecs_test
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/require"
+	"pkg.world.dev/world-engine/cardinal/ecs/component"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"gotest.tools/v3/assert"
+
+	"pkg.world.dev/world-engine/cardinal/ecs"
+	"pkg.world.dev/world-engine/cardinal/ecs/inmem"
+)
+
+type SendEnergyTx struct {
+	From, To string
+	Amount   uint64
+}
+
+type SendEnergyTxResult struct{}
+
+type EnergyComp struct {
+	value int
+}
+
+var Energy = ecs.NewComponentType[EnergyComp]()
+
+func TestWorldLogger(t *testing.T) {
+
+	traceId := "test_trace_id"
+	var buf bytes.Buffer
+	bufLogger := zerolog.New(&buf)
+	//opts is used to hijack the Logger placed in the world object and replace it with something that writes to buf(above)
+	//so it's more easily unit tested. All logging methods called by w.Logger are now written to the buf variable.
+	opt := func(w *ecs.World) {
+		w.Logger = ecs.NewWorldLogger(&bufLogger, w)
+	}
+	w := inmem.NewECSWorldForTest(t, opt)
```

Comment:
Can you just put `w.Logger = ecs.NewWorldLogger(&bufLogger, w)` here without bothering with the opt parameter?

Reply:
yeah. I was going to eventually use this for other test functions, but even that doesn't really improve anything. Will remove.